### PR TITLE
[Keyboard] Add ANAVI Arrows

### PR DESCRIPTION
**Pull request by leon-anavi**
ANAVI Arrows is a compact inverted T hot-swappable mechanical keyboard with 4 Cherry MX compatible switches, translucent key caps, rotary encoder, USB-C, RP2040 microcontroller, backlighting and under lighting.

This commits adds initial QMK support for ANAVI Arrows with:

- Default keymap
- Mini OLED I2C Display
- WS2812B RGB underlighting
- Backlighting (with PWMD5 and channel A as defined for GPIO 26 in the Raspberry Pi RP2040 datasheet)

The Neopixel on XIAO RP2040 is not supported as of the moment.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/anavi/arrows/info.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
+    "keyboard_name": "arrows",
+    "manufacturer": "ANAVI",
+    "url": "https://github.com/AnaviTechnology/anavi-arrows",
+    "maintainer": "leon-anavi",
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "matrix_pins": {
+        "direct": [
+            ["GP27", "GP28", "GP29", "GP0", "GP4"]
```
Comment (zvecr) [inline keyboards/anavi/arrows/info.json:10]:
```suggestion
            ["GP4", "GP0", "GP27", "GP28", "GP29"]
```

File: keyboards/anavi/arrows/info.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    },
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            {"pin_a": "GP1", "pin_b": "GP2", "resolution": 2}
+        ]
+    },
+    "layout_aliases": {
+        "LAYOUT": "LAYOUT_a5"
+    },
```
Comment (zvecr) [inline keyboards/anavi/arrows/info.json:55]:
```suggestion
```

File: keyboards/anavi/arrows/info.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    "layouts": {
+        "LAYOUT_a5": {
+            "layout": [
+                {"matrix": [0, 0], "x": 0, "y": 0},
+                {"matrix": [0, 1], "x": 1, "y": 0},
+                {"matrix": [0, 2], "x": 2, "y": 0},
+                {"matrix": [0, 3], "x": 3, "y": 0},
+                {"matrix": [0, 4], "x": 4, "y": 0}
+            ]
+        }
```
Comment (zvecr) [inline keyboards/anavi/arrows/info.json:65]:
```suggestion
        "LAYOUT": {
            "layout": [
                {"matrix": [0, 0], "x": 2, "y": 0},
                {"matrix": [0, 1], "x": 1, "y": 1},
                {"matrix": [0, 2], "x": 0, "y": 2},
                {"matrix": [0, 3], "x": 1, "y": 2},
                {"matrix": [0, 4], "x": 2, "y": 2}
            ]
        }
```

File: keyboards/anavi/arrows/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _BASE
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BASE] = LAYOUT_a5(
+        KC_RIGHT, KC_DOWN, KC_LEFT, KC_UP, KC_MUTE
```
Comment (zvecr) [inline keyboards/anavi/arrows/keymaps/default/keymap.c:12]:
```suggestion
    [_BASE] = LAYOUT(
                           KC_MUTE,
                  KC_UP,
        KC_RIGHT, KC_DOWN, KC_LEFT
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/anavi/arrows/info.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true,
+        "backlight": true,
+        "oled": true,
+	"combo": true
```
Comment (waffle87) [inline keyboards/anavi/arrows/info.json:23]: This cannot be enabled at the keyboard level.
```suggestion
        "oled": true
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/anavi/arrows/info.json (hunk 5 above)
Comment (leon-anavi) [reply keyboards/anavi/arrows/info.json:23]: OK, thank you. I will move it to `keyboards/anavi/arrows/keymaps/default/rules.mk`

**Comment by leon-anavi** [comment]
@zvecr, @waffle87, thank you for the quick feedback and the code reviews. Following you recommendations I did commit -amend and I have force pushed a new version of the patch.

File: keyboards/anavi/arrows/info.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    },
+    "ws2812": {
+        "pin": "GP3",
+        "driver": "vendor"
+    },
+    "backlight": {
+        "pin": "GP26"
+    },
+    "encoder": {
+        "enabled": true,
```
Comment (waffle87) [inline keyboards/anavi/arrows/info.json:47]:
```suggestion
```

File: keyboards/anavi/arrows/info.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true,
+        "backlight": true,
+        "oled": true
```
Comment (waffle87) [inline keyboards/anavi/arrows/info.json:22]:
```suggestion
        "oled": true,
        "encoder": true
```

**Comment by leon-anavi** [comment]
@waffle87 thanks, I changed `info.json` as you suggested.

**Review by waffle87 — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/anavi/arrows/info.json [hunk 8]
```diff
@@ -0,0 +1,68 @@
+{
+    "keyboard_name": "arrows",
+    "manufacturer": "ANAVI",
+    "url": "https://github.com/AnaviTechnology/anavi-arrows",
+    "maintainer": "leon-anavi",
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "matrix_pins": {
+        "direct": [
+	    ["GP4", "GP0", "GP27", "GP28", "GP29"]
```
Comment (drashna) [inline keyboards/anavi/arrows/info.json:10]:
```suggestion
            ["GP4", "GP0", "GP27", "GP28", "GP29"]
```

**Review by drashna — APPROVED** [review]

**Pull request merged**